### ISSUE TITLE
Provide statically linked binary (CGO_ENABLED=0) to allow smaller docker containers.

### ISSUE THREAD

**Issue by thilomack** [aws/aws-xray-daemon#15]
I would like to reduce the size of my xray daemon docker container.

By compiling the executable myself with "CGO_ENABLED=0" i was able to reduce the size to 12mb.
This is because if the binary already contains all necessary libraries, i do not need a base image to start from. So the docker image only needs to contain the binary and a ca-bundle.

It would be really helpful if you could provide an executable compiled like that with the other executables.

Currently my Dockerfile looks like this:
```
# source for ca-bundle so we can do https requests
FROM amazonlinux:2 as trusted

# build statically linked binary
FROM golang:1.11 as builder
RUN go get -d -u github.com/aws/aws-xray-daemon/...
RUN cd ${GOPATH}/src/github.com/aws/aws-xray-daemon && CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o build/xray ./daemon/daemon.go ./daemon/tracing.go

# start a container from scratch containing only necessary files
FROM scratch
COPY --from=trusted /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
COPY --from=builder /go/src/github.com/aws/aws-xray-daemon/build/xray /xray
EXPOSE 2000
ENTRYPOINT ["/xray","--bind=0.0.0.0:2000"]
```

If the binary i need was available (e.g. as aws-xray-daemon-linux-bundle-3.x.zip) i could reduce the Dockerfile to:
```
FROM amazonlinux:2 as builder
RUN yum install -y unzip
RUN curl -o daemon.zip https://s3.dualstack.us-east-2.amazonaws.com/aws-xray-assets.us-east-2/xray-daemon/aws-xray-daemon-linux-bundle-3.x.zip
RUN unzip daemon.zip && cp xray /xray

# start a container from scratch containing only necessary files
FROM scratch
COPY --from=builder /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
COPY --from=builder /xray /xray
```

**Comment by yogiraj07** [comment]
Hi @thilomack ,
I would add this to feature request. Thank you for the suggestion. 

Best,
Yogi

**Comment by tqh** [comment]
You could even compile it in the build:
```
# source for ca-bundle so we can do https requests
FROM amazonlinux:2 as trusted

# build statically linked binary
FROM golang:1.11.2 as builder
RUN go get -d -u github.com/aws/aws-xray-daemon/...
RUN cd ${GOPATH}/src/github.com/aws/aws-xray-daemon && CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o build/xray ./daemon/daemon.go ./daemon/tracing.go

# start a container from scratch containing only necessary files
FROM scratch
COPY --from=trusted /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
COPY --from=builder /go/src/github.com/aws/aws-xray-daemon/build/xray /xray
EXPOSE 2000
ENTRYPOINT ["/xray","--bind=0.0.0.0:2000"]
```

**Comment by willarmiros** [comment]
@anuraaga I believe this is solved by #84, can you confirm? If so, we can mark it `pending release`.

**Comment by anuraaga** [comment]
Yup this will happen in the next release.

**Comment by srprash** [comment]
Hi @thilomack 
This has been released in v3.3.0 but we recommend using v3.3.1 which includes bug fixes as well. Please see the [changelog](https://github.com/aws/aws-xray-daemon/blob/master/CHANGELOG.md) for details.
